webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +96,776 @@ public WebSiteSessionController createComponentSessionController(
-          // view en cas de rechargement de la page pour naviguer dans le chemin
-          // ou createSite annule
-          // ou upload d'image
+  private String listAllSitesForReader(final WebSiteSessionController scc,
+      final HttpRequest request) throws WebSitesException {
+    String action = request.getParameter("Action");
+    String id = request.getParameter("Id");
 
-          SiteDetail site = scc.getSessionSite();
-          id = site.getPK().getId();
-          request.setAttribute("Site", site);
+    if (action == null) {
+      id = "0";
+      action = "Search";
+    }
 
-        }
+    FolderDetail webSitesCurrentFolder = scc.getFolder(id);
+    scc.setSessionTopic(webSitesCurrentFolder);
+    request.setAttribute(CURRENT_FOLDER_PARAM, webSitesCurrentFolder);
 
-        destination = "/webSites/jsp/design.jsp?Action=design&path=" + currentPath + "&Id=" + id;
-      } else if (function.equals("EffectiveUploadFile")) {
-        List<FileItem> items = request.getFileItems();
-        boolean status = true;
-        String thePath = FileUploadUtil.getParameter(items, "path");
-        scc.checkPath(thePath);
-        FileItem item = FileUploadUtil.getFile(items);
-        if (item != null) {
-          status = (scc.addFileIntoWebSite(thePath, item) == 0);
-        }
+    return "/webSites/jsp/listSite_reader.jsp?Action=" + action + "&Id=" + id;
+  }
 
-        request.setAttribute("UploadOk", status);
+  private String listAllSites(final WebSiteSessionController scc, final HttpRequest request)
+      throws WebSitesException {
+    String action = request.getParameter("Action");
+    String id = request.getParameter("Id");
 
-        destination = "/webSites/jsp/uploadFile.jsp?path=" + thePath;
-      } else if (function.startsWith("descUpload.jsp")) {
-        request.setAttribute("AllIcons", scc.getAllIcons());
-        destination = "/webSites/jsp/descUpload.jsp";
-      } else if (function.equals("EffectiveUploadSiteZip")) {
-        List<FileItem> items = request.getFileItems();
-
-        String nomSite = FileUploadUtil.getParameter(items, "nomSite");
-        String description = FileUploadUtil.getParameter(items, "description");
-        String popupString = FileUploadUtil.getParameter(items, "popup");
-        int popup = 0;
-        if ("on".equals(popupString)) {
-          popup = 1;
-        }
-        String nomPage = FileUploadUtil.getParameter(items, "nomPage");
-        String listeIcones = FileUploadUtil.getParameter(items, "ListeIcones");
-        String listeTopics = FileUploadUtil.getParameter(items, "ListeTopics");
-        String positions = FileUploadUtil.getParameter(items, "Positions");
+    if (action == null) {
+      id = "0";
+      action = "Search";
+    }
 
-        FileItem fileItem = FileUploadUtil.getFile(items);
-        if (fileItem != null) {
-          /* recuperation de l'id = nom du directory */
-          String id = scc.getNextId();
-
-          // Persist uploaded websi
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "Admin" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)